### PR TITLE
Addons authentication and cookie storage

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 3.x
| Description?      | Allow an user to connect to Addons from any BO page. Store cookie after authentication and delete it after logout. Implement PermissionChecker to check if the user has the proper access level
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | This PR can only be tested with https://github.com/PrestaShop/PrestaShop/pull/27485 <br> Go to BO, Module manager or Module catalog, connect to addons from the toolbar button. After authentication, the login button will be replaced by logout button and you'll see the BO cookies `username_addons`, `password_addons` and `is_contributor`. In the other page, you'll also be connected. After logout the cookies disappear and you'll see again the login CTA

### CODE REVIEW

File: src/Controller/Admin/AddonsController.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        $request = $this->requestStack->getCurrentRequest();
+
+        if ($request->isXmlHttpRequest()) {
+            $response = new JsonResponse();
+            $response->setData([
+                'success' => 1,
+                'message' => '',
+            ]);
+        } else {
+            if ($request->server->get('HTTP_REFERER')) {
```
Comment (PierreRambaud) [inline src/Controller/Admin/AddonsController.php:137]: Watch out with HTTP_REFERER, you can forge a request easily: https://web.dev/referrer-best-practices/

File: views/js/addons-connector.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        addonsConnectModalSelector: '#module-modal-addons-connect',
+        addonsLogoutModalSelector: '#module-modal-addons-logout',
+        addonsConnectForm: '#addons-connect-form',
+    };
+
+    var AddonsConnector = function(pageMap) {
+        this.initConnect = function() {
+            // Make addons connect modal ready to be clicked
+            if ($(pageMap.addonsConnectModalBtnSelector).attr('href') === '#') {
+                $(pageMap.addonsConnectModalBtnSelector).attr('data-toggle', 'modal');
```
Comment (PierreRambaud) [inline views/js/addons-connector.js:40]:
```suggestion
                $(pageMap.addonsConnectModalBtnSelector).data('toggle', 'modal');
```
use dedicated method instead using attribute which are not really using the data collection

File: views/js/addons-connector.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                $(pageMap.addonsConnectModalBtnSelector).attr('data-target', pageMap.addonsConnectModalSelector);
+            }
+
+            if ($(pageMap.addonsLogoutModalBtnSelector).attr('href') === '#') {
```
Comment (PierreRambaud) [inline views/js/addons-connector.js:44]: duplicate code

File: src/Controller/Admin/AddonsController.php (hunk 1 above)
Comment (sowbiba) [reply src/Controller/Admin/AddonsController.php:137]: The alternatives given are only acting on server side. Do you have a recommandation/advice ?

File: src/Traits/UseHooks.php [hunk 4]
```diff
@@ -258,6 +262,51 @@ public function hookActionAdminControllerSetMedia(): void
...
+     * This hook adds buttons for the module catalogue toolbar
+     *
+     * @param array $params
+     *
+     * @return array
+     */
+    public function hookActionAdminModuleExtraToolbarButton(array $params)
+    {
+        /** @var DataProvider $addonsProvider */
+        $addonsProvider = $this->get('mbo.addon.module.data_provider.addons');
```
Comment (intraordinaire) [inline src/Traits/UseHooks.php:275]: What is the difference between this and the same logic in [`ModuleCatalogController`](https://github.com/PrestaShopCorp/ps_mbo/pull/127/files#diff-351301f3adf9cb1fcc43b4a2673cc686e98b0a296ac03a439355ac9b1180aa24R224) ? Do we need both ? And if yes, can we factorize this portion of code ?

**Review by intraordinaire — APPROVED** [review]

File: src/Controller/Admin/AddonsController.php [hunk 5]
```diff
@@ -103,7 +103,7 @@ public function loginAction(): JsonResponse
             );
 
             $response->setData(['success' => 1, 'message' => '']);
-            $this->modulesDataProvider->clearCatalogCache();
+//            $this->modulesDataProvider->clearCatalogCache();
```
Comment (sowbiba) [inline src/Controller/Admin/AddonsController.php:106]: This method was cleared from the Core. We need to reimplement that logic

File: views/templates/admin/controllers/module_catalog/includes/card_grid.html.twig [hunk 6]
```diff
@@ -74,7 +74,7 @@
         </div>
         <div class="module-read-more-grid">
           {% if module.attributes.id != "0" %}
-            <a class="module-read-more-grid-btn url" href="{{ path('admin_module_cart', {"moduleId": module.attributes.id }) }}" data-target="#module-modal-read-more-{{module.attributes.name }}{{ additionalModalSuffix|default('') }}">{{ 'Read More'|trans({}, 'Admin.Modules.Feature') }}</a>
+{#            <a class="module-read-more-grid-btn url" href="{{ path('admin_module_cart', {"moduleId": module.attributes.id }) }}" data-target="#module-modal-read-more-{{module.attributes.name }}{{ additionalModalSuffix|default('') }}">{{ 'Read More'|trans({}, 'Admin.Modules.Feature') }}</a>#}
```
Comment (sowbiba) [inline views/templates/admin/controllers/module_catalog/includes/card_grid.html.twig:77]: Route `admin_module_cart` doesn't exist anymore in the Core. We need to reimplement or get rid of this

**Pull request merged**